crates/sozo/ops/src/migration/mod.rs

```diff
@@ -174,3 +180,87 @@ enum ContractDeploymentOutput {
     Output(UpgradeOutput),
 }
+
+pub fn generate_overlays(ws: &Workspace<'_>) -> Result<()> {
+    let profile_name =
+        ws.current_profile().expect("Scarb profile expected to be defined.").to_string();
+
+    // its path to a file so `parent` should never return `None`
+    let manifest_dir = ws.manifest_path().parent().unwrap().to_path_buf();
+    let profile_dir = manifest_dir.join(MANIFESTS_DIR).join(profile_name);
+
+    let base_manifests = profile_dir.join(BASE_DIR);
+
+    let world = OverlayClass { name: WORLD_CONTRACT_NAME.into(), original_class_hash: None };
+    let base = OverlayClass { name: BASE_CONTRACT_NAME.into(), original_class_hash: None };
+
+    // generate default OverlayManifest from base manifests
+    let contracts = overlay_dojo_contracts_from_path(&base_manifests.join(CONTRACTS_DIR))
+        .with_context(|| "Failed to build default DojoContract Overlays from path.")?;
+    let models = overlay_model_from_path(&base_manifests.join(MODELS_DIR))
+        .with_context(|| "Failed to build default DojoModel Overlays from path.")?;
+
+    let default_overlay =
+        OverlayManifest { world: Some(world), base: Some(base), contracts, models };
+
+    let overlay_path = profile_dir.join(OVERLAYS_DIR);
+
+    // read existing OverlayManifest from path
+    let mut overlay_manifest = OverlayManifest::load_from_path(&overlay_path)
+        .with_context(|| "Failed to load OverlayManifest from path.")?;
+
+    // merge them to get OverlayManifest which contains all the contracts and models from base
+    // manifests
+    overlay_manifest.merge(default_overlay);
+
+    overlay_manifest
+        .write_to_path_nested(&overlay_path)
+        .with_context(|| "Failed to write OverlayManifest to path.")?;
+
+    Ok(())
+}
+
+fn overlay_dojo_contracts_from_path(path: &Utf8PathBuf) -> Result<Vec<OverlayDojoContract>> {
+    let mut elements = vec![];
+
+    let entries = path
+        .read_dir()?
+        .map(|entry| entry.map(|e| e.path()))
+        .collect::<Result<Vec<_>, io::Error>>()?;
+
+    for path in entries {
+        if path.is_file() {
+            let manifest: Manifest<DojoContract> = toml::from_str(&fs::read_to_string(path)?)?;
+
+            let overlay_manifest =
+                OverlayDojoContract { name: manifest.name, ..Default::default() };
+            elements.push(overlay_manifest);
+        } else {
+            continue;
+        }
+    }
+
+    Ok(elements)
+}
+
+fn overlay_model_from_path(path: &Utf8PathBuf) -> Result<Vec<OverlayDojoModel>> {
+    let mut elements = vec![];
+
+    let entries = path
+        .read_dir()?
+        .map(|entry| entry.map(|e| e.path()))
+        .collect::<Result<Vec<_>, io::Error>>()?;
+
+    for path in entries {
+        if path.is_file() {
+            let manifest: Manifest<DojoModel> = toml::from_str(&fs::read_to_string(path)?)?;
+
+            let overlay_manifest = OverlayDojoModel { name: manifest.name, ..Default::default() };
```

Comment:
If we can't inject a commented line with all possible entries, do you think we could have default values?
If not no problem, we should ensure user can rely on updated doc. 👍

Reply:
i thought about it but currently we override the value based on if its set in overlays or not. to do this we would need to merge based on if the value is "default" or not, but there would be ambiguity of what we should do if overlay has default value and base manifest has some non default value so it would complicate things.

Reply:
Sounds good, let's keep it simple for now, thanks for the update mate. 👍